src/contrib/persistence/Akka.Persistence.Sql.Linq2Db/Journal/Linq2DbWriteJournal.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.Persistence.Journal;
+using Akka.Persistence.Sql.Linq2Db.Config;
+using Akka.Persistence.Sql.Linq2Db.Db;
+using Akka.Persistence.Sql.Linq2Db.Journal.DAO;
+using Akka.Persistence.Sql.Linq2Db.Journal.Types;
+using Akka.Persistence.Sql.Linq2Db.Utility;
+using Akka.Streams;
+using Akka.Streams.Dsl;
+using Akka.Util;
+using Akka.Util.Internal;
+
+namespace Akka.Persistence.Sql.Linq2Db.Journal
+{
+    public class DateTimeHelpers
+    {
+        private static DateTime UnixEpoch = new DateTime(1970,1,1,0,0,0,DateTimeKind.Utc);
+
+        public static long ToUnixEpochMillis(DateTime time)
+        {
+            long unixTime =
+                (long) (time.ToUniversalTime() - UnixEpoch).TotalMilliseconds;
+            return unixTime;
+        }
+        public static long UnixEpochMillis()
+        {
+            long currentTime =
+                (long) (DateTime.UtcNow - UnixEpoch).TotalMilliseconds;
+            return currentTime;
+        }
+
+        public static DateTime FromUnixEpochMillis(in long unixEpochMillis)
+        {
+            return UnixEpoch.AddMilliseconds(unixEpochMillis);
+        }
+    }
+
+    public class Linq2DbWriteJournal : AsyncWriteJournal
+    {
+        public static Configuration.Config DefaultConfiguration =>
+            ConfigurationFactory.FromResource<Linq2DbWriteJournal>(
+                "Akka.Persistence.Sql.Linq2Db.persistence.conf");
+        
+        private ActorMaterializer _mat;
+        private JournalConfig _journalConfig;
+        private ByteArrayJournalDao _journal;
+        public Linq2DbWriteJournal(Configuration.Config config)
+        {
+            try
+            {
+                _journalConfig = new JournalConfig(config);
+                _mat = Materializer.CreateSystemMaterializer((ExtendedActorSystem)Context.System,
+                    ActorMaterializerSettings.Create(Context.System)
+                        .WithDispatcher(_journalConfig.MaterializerDispatcher)
+                    ,
+                    "l2dbWriteJournal"
+                );
```

Comment:
Since persistence plugins run under /system/ , we should run Materializer there as well; otherwise shutdowns can get messy (shutting /user/ down first will kill the stream.)